src/pygama/dsp/_processors/pole_zero.py

```diff
@@ -80,15 +83,20 @@ def double_pole_zero(w_in, t_tau1, t_tau2, frac, w_out):
 
     if np.isnan(w_in).any() or np.isnan(t_tau1) or np.isnan(t_tau2) or np.isnan(frac):
         return
+    if (len(w_in)<=3):
+        raise DSPFatal('The length of the waveform must be larger than 3 for the filter to work safely')
+
+    a = np.exp(-1 / t_tau1)
+    b = np.exp(-1 / t_tau2)
+
+    transfer_denom_1 = -1*(frac*b-frac*a-b-1)
+    transfer_denom_2 = frac*b-frac*a-b
+    transfer_num_1 = -1*(a+b)
+    transfer_num_2 = a*b
 
-    const1 = 1 / t_tau1
-    const2 = 1 / t_tau2
     w_out[0] = w_in[0]
-    e1 = w_in[0]
-    e2 = w_in[0]
-    e3 = 0
-    for i in range(1, len(w_in), 1):
-        e1 += w_in[i] - e2 + e2 * const1
-        e3 += w_in[i] - e2 - e3 * const2
-        e2  = w_in[i]
-        w_out[i] = e1 - frac * e3
+    w_out[1] = w_in[1]
+    w_out[2] = w_in[2]
+
+    for i in range(2, len(w_in), 1):
+        w_out[i] = transfer_denom_1*w_out[i-1] + transfer_denom_2*w_out[i-2] + w_in[i] + transfer_num_1*w_in[i-1] + transfer_num_2*w_in[i-2]
```

Comment:
One last thing @SamuelBorden -- I think rewriting this line as follows would better elucidate the structure of the linear convolution algorithm (note the negative signs for denom's, requiring changes where they are defined above). What do you think? 
`w_out[i] = w_in[i] + transfer_num_1*w_in[i-1] + transfer_num_2*w_in[i-2] - transfer_denom_1*w_out[i-1] - transfer_denom_2*w_out[i-2] + `